From 08a267525600277a6073f7044c1297ff4744ea48 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <catap@catap.ru>
Date: Tue, 2 Nov 2021 18:40:01 +0000
Subject: [PATCH] Fixed support macOS before 10.12 Sierra

macOS before 10.12 Sierra hasn't got defined `MAC_OS_X_VERSION_10_12`.

As result if always true because any `MAC_OS_X_VERSION_MIN_REQUIRED`
is bigger than nothing
---
 reflink/darwin.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/reflink/darwin.c b/reflink/darwin.c
index 6714d62..cc2489a 100644
--- a/reflink/darwin.c
+++ b/reflink/darwin.c
@@ -1,6 +1,6 @@
 #include <AvailabilityMacros.h>
 /* NOTE: sys/clonefile.h is available since OS X 10.12 */
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12
+#if defined MAC_OS_X_VERSION_10_12 && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12
 #include <sys/clonefile.h> /* for clonefile(2) */
 #include <errno.h>
 
-- 
GitLab

